### DISCUSSION TITLE
Adding a Java 21 tag that supports ARMv7

### DISCUSSION THREAD

**Discussion by Restfulbee37** [itzg/docker-minecraft-server#2755] [Ideas]
Hey,

I was just wondering with the latest snapshot (24W14A) now requiring Java 21 (as referenced here: https://www.minecraft.net/en-us/article/minecraft-snapshot-24w14a) if you were in the process of developing a docker tag that will support Java 21 for ARMv7 and if so do you have any rough ETA?

I say this as I have a MC server running on my Pi and now can't access any of my worlds.

Thanks :)

**Comment by itzg** [comment]
Unfortunately the base image used for Java 21 doesn't provide armv7 support

https://hub.docker.com/_/eclipse-temurin/tags?page=1&name=21-jre

**Reply by Restfulbee37** [reply]
Ah okay that makes sense, I do see that it supports arm64/v8 for java 21 which is supported on the pi4 and 5, do you plan on making a version for arm64/v8 to futureproof for when MC full version inevitably switches to 64 bit only and not just the snapshot?

If not that's absolutely fine I'm just curious :)

Thanks

**Reply by itzg** [reply]
The java21 image already picks up that arm64 (and amd64) support

That is also shown in architecture column here

https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/

**Reply by Restfulbee37** [reply]
Thank you :)

**Reply by 3XC1T3D** [reply]
Hi according to that page (https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/) stable/latest supports armv7 .

Unfortunatley there is no armv7 image - which image should i use on an pi4 ?

Best regards

**Reply by itzg** [reply]
> Hi according to that page (https://docker-minecraft-server.readthedocs.io/en/latest/versions/java/) stable/latest supports armv7 .

Sorry, that was a mistake in the docs when latest/stable for shifted to Java 21. 

As mentioned I'm limited by the base image support for Java versions and architecture types. So on older RaspberryPi you're limited to Java 17 and older.

**Comment by itzg** [comment]
Noting the Adoptium/Temurin issue where armv7/arm32 is discussed

https://github.com/adoptium/adoptium-support/issues/962
